.github/workflows/bandit-scan.yml

```
@@ -0,0 +1,21 @@
+name: Bandit Code Scan
+
+on:
+  pull_request:
+    branches: [ main ]
+
+permissions:
+    pull-requests: write
+jobs:
+  bandit-action:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```

The latest version for checkout is v4

```suggestion
    - uses: actions/checkout@v4
```